### PR TITLE
ascanrules: use timing utils in Hyper scan rule

### PR DESCRIPTION
**Pull request by thc202**
Adapt `SqlInjectionHypersonicScanRule` to do the time-based tests with the timing utils.

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

**Review by kingthorin — APPROVED** [review]

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionHypersonicScanRule.java [hunk 1]
```diff
@@ -267,146 +268,90 @@ public void init() {
...
+
+                        String finalPayload =
+                                sleepPayload
+                                        .replace(ORIG_VALUE_TOKEN, paramValue)
+                                        // Time in milliseconds for the SQL function.
+                                        .replace(SLEEP_TOKEN, Integer.toString(((int) x) * 1000));
+
+                        setParameter(msg, paramName, finalPayload);
+                        LOGGER.debug("Testing [{}] = [{}]", paramName, finalPayload);
+                        attack.compareAndSet(null, finalPayload);
```
Comment (itsdivyanshjain) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionHypersonicScanRule.java:297]: attack variable will always contains the first payload, even though attack payload is actually different other than 1st. I've confirmed this from my local testing.
@thc202

Comment (itsdivyanshjain) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionHypersonicScanRule.java:297]: Even though it is merged, looking forward to get this resolved.

Comment (thc202) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionHypersonicScanRule.java:297]: Right, the `AtomicReference`s should be declared inside the loop.

Comment (thc202) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionHypersonicScanRule.java:297]: #5163

**Pull request merged**